core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -182,7 +182,9 @@ private OAuth2Credentials parseCredentials(HttpRequestInitializer source) {
       Credential credential = (Credential) source;
       AccessToken accessToken = new AccessToken(
           credential.getAccessToken(),
-          new Date(credential.getExpirationTimeMilliseconds())
+          credential.getExpirationTimeMilliseconds() != null
+              ? new Date(credential.getExpirationTimeMilliseconds())
+              : null
```

Comment:
can we use an optional here instead of this ternary? 

something like:

```
Optional<Long> expTime = Optional.ofNullable(credential.getExpirationTimeMilliseconds());

AccessToken accessToken = new AccessToken(
          credential.getAccessToken(),
          expTime.isPresent()
              ? new Date(expTime.get())
              : null
          );
```